### ISSUE TITLE
CI Fails After upgrading UMD submodule to latest

### ISSUE THREAD

**Issue by blozano-tt** [tenstorrent/tt-metal#13618]
post commit fails on https://github.com/tenstorrent/tt-metal/tree/blozano/bump_umd_0
ONLY for N300 machines.

When UMD submodule is pushed to the latest.

So far, both @joelsmithTT and @blozano-tt cannot reproduce locally.

Error message:
```
---- ttSiliconDevice::init_hugepage: bind_area_to_memory_nodeset() failed (physical_device_id: 0 ch: 0). Hugepage allocation is not on NumaNode matching TT Device. Side-Effect is decreased Device->Host perf (Issue #893).
UMD: Mapping a hugepage failed. (device: 0, 1/2 errno: Cannot allocate memory).

File /proc/cmdline  is: 
BOOT_IMAGE=/boot/vmlinuz-5.4.0-196-generic root=UUID=22ef[31](https://github.com/tenstorrent/tt-metal/actions/runs/11240100607/job/31249270233#step:6:32)3e-6adb-4681-a599-4fd0593fda6b ro console=tty1 console=ttyS0 hugepagesz=1G hugepages=2 iommu=pt

File /sys/kernel/mm/hugepages/hugepages-1048576kB/nr_hugepages  is: 
2
2024-10-08 17:14:[32](https://github.com/tenstorrent/tt-metal/actions/runs/11240100607/job/31249270233#step:6:33).834 | FATAL    | Always          - Hugepages must be successfully initialized if workload contains remote chips!
libc++abi: terminating due to uncaught exception of type std::runtime_error: Hugepages must be successfully initialized if workload contains remote chips!
backtrace:
 --- /home/ubuntu/actions-runner/_work/tt-metal/tt-metal/build/lib/libdevice.so(+0x3dea2) [0x7f303f[34](https://github.com/tenstorrent/tt-metal/actions/runs/11240100607/job/31249270233#step:6:35)eea2]
```

Need advice to progress
cc: @tt-rkim @ttmchiou

**Comment by broskoTT** [comment]
Unrelated to this issue, while working on UMD, I hit the same error, so was forced to investigate further. This could be the same issue, or it may be something else given the ambiguity of error message "Cannot allocate memory".

TLDR;
To confirm the issue: run `ls -la /dev/hugepages-1G/`, you'll see some number of files (number specific for this system, probably 4 due to single PCI slot used) with 1GB size, and one or more with 0 size.
To fix: `rm /dev/hugepages-1G/*`

Longer version:
Hugepage mounts work so that a folder is designated by the system where hugepages will live. In this case, it is /dev/hugepages-1G. You would see an entry similar to `hugetlbfs /dev/hugepages-1G hugetlbfs rw,relatime,mode=777,pagesize=1024M 0 0` after executing `cat /proc/mounts | grep huge`.
Now, whenever any process creates a file in that folder, it will be automatically mapped to a hugepage. And it will happen as long as there are available hugepages on the system. When max limit of hugepages has been hit, this will result in file being created of size 0 which will not be mapped to any hugepage. UMD regardless tries to map it as if it was full size, and that part fails. The reason why there are more files (more unique filenames) than hugepages is likely due to some naming scheme being changed in UMD, or some bug in UMD naming these files. We should probably investigate this further. 
However, a simple fix is to just cleanup the folder, which will release back the hugepages to the system.

**Comment by broskoTT** [comment]
Actually, @joelsmithTT found the bug in UMD in naming, we will revert naming to previous scheme. Just wait for the new UMD version with this fix

**Comment by tt-rkim** [comment]
Ah so this was a UMD problem? How come we didn't hit this before? Does the system not automatically cleanup pages that are no longer used by active processes?

**Comment by broskoTT** [comment]
This is hit:
- On a system which old UMD ran, so not a clean system
- On a physical device != 0
- Where >1 channels were used
- Where os was setup at the right number needed by the UMD, so there were no additional hugepages available

The system does not automatically clean these up

**Comment by tt-rkim** [comment]
Hmm... our VMs only have one device mapped into it. But who knows

Anyway, hopefully whatever we were seeing also goes away.

**Comment by tt-rkim** [comment]
Is there an issue on umd side tracking this?

**Comment by joelsmithTT** [comment]
I don't think so but I'm about to push a fix.

**Comment by joelsmithTT** [comment]
https://github.com/tenstorrent/tt-umd/pull/169

**Comment by tt-rkim** [comment]
@blozano-tt did you get what you needed from this umd bump?

**Comment by blozano-tt** [comment]
@tt-rkim I want even more, but this bump gave me a lot.

1. No more crash in CI with new UMD
2. No NEED for ARCH_NAME when building UMD library code.

**Comment by tt-rkim** [comment]
Great!
